decidim/decidim
# Replace social-share-button gem with social share services API

#### :tophat: What? Why?

After v0.25, the Social Share Button gem stopped working, as it isn't compatible with Webpacker. See #8956.
We have mitigated this bug by compiling the code and serving it ourselves (see #9041), but that was a temporary workaround.

This PR implements its functionality, so we don't depend on this external dependency. I've extracted from them:

* the social network icons images
* the sharing URLs
* the CSS, so we can keep the same look and feel

I've started by just adding a prefix to honor the same API, so it was easier to upgrade, but then I started having doubts, because we're trying to drop this direct configuration through initializer (see #8725), so I decided that it'd be better to just make the jump and change it to the ENV VARS -> config/secrets.yml -> decidim initializer pattern that we're migrating to.

Some notable changes in comparison with the old functionality:

* It's all implemented with Ruby/HTML, no JavaScript 😄 
* The configuration for the enabled social networks services is done through our decidim initializer, as I've mentioned
* There's a new Registry, allowing implementers to define new services in their app
* When clicking, as it's a regular external link, it kicks the External Website modal. Before as it was opening through JS it skips it.
* Regarding the Social Networking Services, a couple changed a bit:
  * delicious: it's dead, so we dropped it (https://del.icio.us/)
  * Weibo and QQ: I didn't implement them because I'd need to add an app key to the contract, and they're actually not popular in any country that's using Decidim (they've both from China). I don't have access to any account to test. It shouldn't be difficult to implement them with the Registry API, though.
 
#### :pushpin: Related Issues

- Related to #8956 

#### Testing

1. Add the relevant configuration in your development_app or just generate a new one
2. Go to a resource that has the social sharing option 
3. Click on "Share" button
4. See that the modal is the same
5. Click in any service
6. See that it works

### :camera: Screenshots



:hearts: Thank you!
